cmd/coap/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	coap "github.com/dustin/go-coap"
+	"github.com/mainflux/mainflux/coap"
+	"github.com/mainflux/mainflux/coap/nats"
+
+	broker "github.com/nats-io/go-nats"
+	"go.uber.org/zap"
+)
+
+const (
+	port       int    = 5683
+	defNatsURL string = broker.DefaultURL
+	envNatsURL string = "COAP_ADAPTER_NATS_URL"
+)
+
+type config struct {
+	Port    int
+	NatsURL string
+}
+
+var logger *zap.Logger = nil
```

Comment:
Avoid this, it is an unnecessary initialization. We have this initialization on 2 more places in normalizer service as well, and it should be fixed completely.

Reply:
How should we pass logger instance to the other modules? Through function parameter?